src/identity/configuration/IdentityProviderFactory.ts

```diff
@@ -144,8 +144,23 @@ export class IdentityProviderFactory implements ProviderFactory {
     // Render errors with our own error handler
     this.configureErrors(config);
 
-    // Allow provider to interpret reverse proxy headers
-    const provider = new Provider(this.baseUrl, config);
+    // As oidc-provider is an ESM package and CSS is CJS, we have to use a dynamic import here.
+    // Unfortunately there is a Node/Jest bug that causes segmentation faults when doing such an import in Jest:
+    // https://github.com/nodejs/node/issues/35889
+    // To work around that we do the import differently in case we are in a Jest test run.
+    // This can be detected base on the env variables: https://jestjs.io/docs/environment-variables.
```

Comment:
```suggestion
    // To work around that, we do the import differently, in case we are in a Jest test run.
    // This can be detected via the env variables: https://jestjs.io/docs/environment-variables.
```